cmd/corimDisplay.go

```diff
@@ -61,72 +61,119 @@ func checkCorimDisplayArgs() error {
-		return fmt.Errorf("error loading signed CoRIM from %s: %w", signedCorimFile, err)
+func displaySignedCorim(s corim.SignedCorim, corimFile string, showTags bool) error {
+	metaJSON, err := json.MarshalIndent(&s.Meta, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error encoding CoRIM Meta from %s: %w", corimFile, err)
 	}
 
-	if err = s.FromCOSE(signedCorimCBOR); err != nil {
-		return fmt.Errorf("error decoding signed CoRIM from %s: %w", signedCorimFile, err)
+	fmt.Println("Meta:")
+	fmt.Println(string(metaJSON))
+
+	corimJSON, err := json.MarshalIndent(&s.UnsignedCorim, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error encoding unsigned CoRIM from %s: %w", corimFile, err)
 	}
 
-	if metaJSON, err = json.MarshalIndent(&s.Meta, "", "  "); err != nil {
-		return fmt.Errorf("error decoding CoRIM Meta from %s: %w", signedCorimFile, err)
+	fmt.Println("Corim:")
+	fmt.Println(string(corimJSON))
+
+	if showTags {
+		fmt.Println("Tags:")
+		// convert []corim.Tag to [][]byte
+		tags := make([][]byte, len(s.UnsignedCorim.Tags))
+		for i, tag := range s.UnsignedCorim.Tags {
+			tags[i] = tag
+		}
+		displayTags(tags)
 	}
 
-	fmt.Println("Meta:")
-	fmt.Println(string(metaJSON))
+	return nil
+}
 
-	if corimJSON, err = json.MarshalIndent(&s.UnsignedCorim, "", "  "); err != nil {
-		return fmt.Errorf("error decoding unsigned CoRIM from %s: %w", signedCorimFile, err)
+func displayUnsignedCorim(u corim.UnsignedCorim, corimFile string, showTags bool) error {
+	corimJSON, err := json.MarshalIndent(&u, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error encoding unsigned CoRIM from %s: %w", corimFile, err)
 	}
 
 	fmt.Println("Corim:")
 	fmt.Println(string(corimJSON))
 
 	if showTags {
 		fmt.Println("Tags:")
-		for i, e := range s.UnsignedCorim.Tags {
-			// need at least 3 bytes for the tag and 1 for the smallest bstr
-			if len(e) < 3+1 {
-				fmt.Printf(">> skipping malformed tag at index %d\n", i)
-				continue
-			}
-
-			// split tag from data
-			cborTag, cborData := e[:3], e[3:]
-
-			hdr := fmt.Sprintf(">> [ %d ]", i)
-
-			if bytes.Equal(cborTag, corim.ComidTag) {
-				if err = printComid(cborData, hdr); err != nil {
-					fmt.Printf(">> skipping malformed CoMID tag at index %d: %v\n", i, err)
-				}
-			} else if bytes.Equal(cborTag, corim.CoswidTag) {
-				if err = printCoswid(cborData, hdr); err != nil {
-					fmt.Printf(">> skipping malformed CoSWID tag at index %d: %v\n", i, err)
-				}
-			} else if bytes.Equal(cborTag, cots.CotsTag) {
-				if err = printCots(cborData, hdr); err != nil {
-					fmt.Printf(">> skipping malformed CoTS tag at index %d: %v\n", i, err)
-				}
-			} else {
-				fmt.Printf(">> unmatched CBOR tag: %x\n", cborTag)
-			}
+		// convert []corim.Tag to [][]byte
+		tags := make([][]byte, len(u.Tags))
+		for i, tag := range u.Tags {
+			tags[i] = tag
 		}
+		displayTags(tags)
 	}
 
 	return nil
 }
 
+func display(corimFile string, showTags bool) error {
```

Comment:
Generally you’d want a rendering function to be version in input and output types, so I’d recommend reading from an io.Reader and writing to an io.Writer. You can then make convenience functions that prepare the options for you. In terms of options, showTags as a boolean input makes calling positions hard to interpret and makes adding options more difficult. I’d recommend making the writer and showTags options fields of a DisplayOptions struct to take as the only other argument from the reader.

Reply:
@deeglaze I appreciate your recommendation to utilize io.Reader and io.Writer along with a DisplayOptions struct to enhance the versatility and flexibility of our rendering functions.

However, I’ve observed that the majority of the code in our repository currently follows a different approach. To maintain consistency throughout the project, I believe it would be more effective to adhere to our existing coding patterns.  

Could you please share some references or examples in contexts similar to our project? This would help me better understand how we might integrate these improvements in a way that aligns with our current structure...

Reply:
@deeglaze may be i could raise a new issue about changing to this pattern all the instances where there is io kinda situation